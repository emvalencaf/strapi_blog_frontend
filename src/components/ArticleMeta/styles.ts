import styled, { css } from "styled-components";

export const Wrapper = styled.footer`
	${({ theme }) => css`
		font-size: ${theme.font.sizes.small};
		color: ${theme.colors.darkerGray};

		.categories span::after {
			content: ", ";
		}

		.categories span:last-child::after {
			content: "";
		}

		a {
			color: ${theme.colors.secondary};
			text-decoration: none;
			transition: : all 350ms ease-in-out;

			&:hover {
				filter: brightness(50%);
			}
		}


	`}
`;
