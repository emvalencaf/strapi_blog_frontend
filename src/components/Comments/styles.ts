import styled, { css } from "styled-components";

export const Wrapper = styled.div`
	${({ theme }) => css`
		max-width: ${theme.sizes.content};
		width: 100%;
		margin: ${theme.spacings.large} 0;
		padding: 0 ${theme.spacings.large};
	`}
`;
