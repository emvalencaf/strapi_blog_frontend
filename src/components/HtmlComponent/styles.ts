import styled, { css } from "styled-components";

export const Container = styled.div`
	${({ theme }) => css`
    @media ${theme.media.lteMedium} {
      font-size: 2rem;

      .image-style-side {
        max-width: 100%;
        float: none;
      }
    }

    font-size: ${theme.font.sizes.medium};


    p {
      margin: ${theme.spacings.xlarge} 0;
    }

    a, a:visited, a:link {
      color: ${theme.colors.secondary};
      text-decoration: none;
      transition: all 350ms ease-in-out;
    }

    a:hover {
      filter: brightness(50%);
    }

    pre {
      font-size: ${theme.font.sizes.small};
      background-color: ${theme.colors.primary};
      padding: ${theme.spacings.medium};
      font-family: monospace;
      color: ${theme.colors.white}
      margin: ${theme.spacings.xlarge};
      width: 100%;
      overflow-x: auto;
    }

    .image {
      background: ${theme.colors.mediumGray};
      line-height: 0;
      margin: ${theme.spacings.xlarge} 0;
    }

    .image figcaption {
      font-size: ${theme.font.sizes.small};
      padding: ${theme.spacings.small};
      text-align: center;
      line-height: 1.3;
    }


    .image-style-side {
      max-width: 50%;
      float: right;
      margin: ${theme.spacings.medium};
    }

    hr {
      border: none;
      border-bottom: 0.1rem solid $${theme.colors.mediumGray};
    }

    ul, ol {
      margin: ${theme.spacings.xlarge};
    }

    .table {
      width: 100%;
      overflow: hidden;
      overflow-x: auto;

    }

    table {
      width: 100%;
      border-collapse: collapse;
    }

    .table td, .table th {
      padding: ${theme.spacings.small};
      border: 0.1rem solid ${theme.colors.mediumGray};
    }
  `}
`;
