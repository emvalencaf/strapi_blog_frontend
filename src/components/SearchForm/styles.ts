import styled, { css } from "styled-components";

export const SearchContainer = styled.div`
	${({ theme }) => css`
		margin: 0 auto;
		margin-bottom: ${theme.spacings.xlarge};
		padding: 0 ${theme.spacings.large};
		max-width: 120rem;
		width: 100%;
		display: flex;
		align-items: center;
		justify-content: center;

		& form {
			display: flex;
			width: 80%;
			max-width: 800px;
			border: 2px solid ${theme.colors.darkerGray};
			border-radius: 2rem;
		}
	`}
`;

export const SearchInput = styled.input`
	${({ theme }) => css`
		padding: 0.5em ${theme.spacings.small};
		width: 100%;
		border: none;
		border-radius: 20px;
		background: transparent;

		&:focus {
			outline: none;
		}
	`}
`;

export const SearchButton = styled.button`
	${({ theme }) => css`
		width: 3rem;
		height: 3rem;
		border: none;
		padding: 0 0.2rem;
		border-left: 2px solid ${theme.colors.darkerGray};
		background: transparent;
		cursor: pointer;
		transition: all 350ms ease-in-out;
		color: inherit;
		&:hover {
			color: ${theme.colors.secondary};
		}
	`}
`;
