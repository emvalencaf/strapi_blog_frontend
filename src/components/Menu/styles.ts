import styled, { css, DefaultTheme } from "styled-components";

// imported styled
import { Title as HeadingStyles } from "../Heading/styles";

// types
type MenuBehaviorParams = {
	menuVisible: boolean;
	theme: DefaultTheme;
};

const wrapperChanger = (menuVisible: boolean, theme: DefaultTheme) => css`
	@media ${theme.media.lteMedium} {
		left: ${menuVisible ? "0" : "-32rem"};
	}

	left: ${menuVisible ? "0" : "-30rem"};
	overflow-y: ${menuVisible ? "auto" : "hidden"};
`;

const buttonChanger = (menuVisible: boolean, theme: DefaultTheme) => css`
	@media ${theme.media.lteMedium} {
		left: ${menuVisible ? "26rem" : "1rem"};
	}

	left: ${menuVisible ? "26rem" : "-1rem"};
	color: ${menuVisible ? theme.colors.secondary : theme.colors.white};
`;

export const Wrapper = styled.div<MenuBehaviorParams>`
	${({ theme, menuVisible }) => css`
		background: ${theme.colors.primary};
		padding: ${theme.spacings.large};
		display: flex;
		position: fixed;
		z-index: 1;
		width: 100%;
		max-width: 32rem;
		height: 100vh;
		top: 0;
		left: 0;
		transition: all 350ms ease-in-out;
		${wrapperChanger(menuVisible, theme)}
	`}
`;

export const Nav = styled.nav`
	${() => css`
		margin: auto;
		width: 100%;
	`}
`;

export const Logo = styled.div`
	${({ theme }) => css`
		${HeadingStyles} {
			display: flex;
			justify-content: center;
			margin: 0;
			margin-bottom: ${theme.spacings.xlarge};

			img {
				border: 0.5rem solid ${theme.colors.secondary};
			}
		}
	`}
`;

export const OpenClose = styled.a<MenuBehaviorParams>`
	${({ theme, menuVisible }) => css`
		position: fixed;
		top: ${theme.spacings.medium};
		color: ${theme.colors.white};
		background: ${theme.colors.primary};
		z-index: 2;
		width: 3rem;
		height: 3rem;
		left: 26rem;
		transition: all 350ms ease-in-out;
		${buttonChanger(menuVisible, theme)}
	`}
`;
