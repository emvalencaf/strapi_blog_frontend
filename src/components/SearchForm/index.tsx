// icons
import { SearchAlt } from "@styled-icons/boxicons-regular";
import { Cancel, CheckCircleOutline } from "@styled-icons/material-outlined";

// styles
import * as Styled from "./styles";

// types
import { ChangeEvent } from "react";
export type SearchFormProps = {
	searchValue?: string | string[];
	handleChange: (event: ChangeEvent<HTMLInputElement>) => void;
	disabled?: boolean;
};

const SearchForm = ({
	searchValue = "",
	handleChange,
	disabled = false,
}: SearchFormProps) => {
	return (
		<Styled.SearchContainer>
			<div>
				<Styled.SearchInput
					type="search"
					placeholder="encontre posts"
					name="q"
					value={searchValue}
					onChange={handleChange}
					disabled={disabled}
				/>
				{disabled ? (
					<Cancel
						className="search-cancel-icon"
						aria-label="Input disabled"
					/>
				) : (
					<CheckCircleOutline
						className={"search-ok-icon"}
						aria-label="Input enabled"
					/>
				)}
				<Styled.SearchButton>
					<SearchAlt aria-label="search button" />
				</Styled.SearchButton>
			</div>
		</Styled.SearchContainer>
	);
};

export default SearchForm;
