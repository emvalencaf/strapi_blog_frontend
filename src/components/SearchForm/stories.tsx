import { Meta, Story } from "@storybook/react/types-6-0";
import SearchForm, { SearchFormProps } from ".";

export default {
	title: "SearchForm",
	component: SearchForm,
	args: {
		query: "query",
	},
	argTypes: {
		query: { type: "string" },
	},
} as Meta<SearchFormProps>;

export const Template: Story<SearchFormProps> = () => {
	return (
		<div>
			<SearchForm />
		</div>
	);
};
