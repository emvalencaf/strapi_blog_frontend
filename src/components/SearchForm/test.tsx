import { renderTheme } from "../../styles/render-theme";

// react component to be test
import SearchForm from ".";
import { fireEvent, logDOM, screen } from "@testing-library/react";

describe("<SearchForm />", () => {
	it("should fired component handleChange", () => {
		const setSearch = jest.fn((v) => v);
		renderTheme(<SearchForm handleChange={setSearch} />);

		const input = screen.queryByPlaceholderText("encontre posts");
		fireEvent.change(input, { target: { value: "test" } });

		expect(setSearch).toHaveBeenCalled();
	});

	it("should render a cancel-icon within the input", () => {
		const setSearch = jest.fn();
		renderTheme(<SearchForm handleChange={setSearch} disabled />);

		expect(screen.getByLabelText("Input disabled")).toBeInTheDocument();
	});

	it("should match a snapshot", () => {
		const setSearch = jest.fn();
		const { container } = renderTheme(
			<SearchForm handleChange={setSearch} />
		);

		expect(container).toMatchSnapshot();
	});
});
