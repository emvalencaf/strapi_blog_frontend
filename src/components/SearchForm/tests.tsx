import { renderTheme } from "../../styles/render-theme";

// react component to be test
import SearchForm from ".";

describe("<SearchForm />", () => {
	it("should match a snapshot", () => {
		const { container } = renderTheme(<SearchForm />);

		expect(container).toMatchSnapshot();
	});
});
