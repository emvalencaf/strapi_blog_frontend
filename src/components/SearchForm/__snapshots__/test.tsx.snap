// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SearchForm /> should match a snapshot 1`] = `
.c2 {
  display: inline-block;
  vertical-align: middle;
  overflow: hidden;
}

.c0 {
  margin: 0 auto;
  margin-bottom: 4.0rem;
  padding: 0 3.2rem;
  max-width: 120rem;
  width: 100%;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
}

.c0 div {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  width: 80%;
  max-width: 800px;
  border: 2px solid #AAAAAA;
  border-radius: 2rem;
}

.c0 div svg.search-cancel-icon,
.c0 div svg.search-ok-icon {
  -webkit-align-self: center;
  -ms-flex-item-align: center;
  align-self: center;
  justify-self: center;
  height: 2rem;
  width: 2rem;
}

.c0 div svg.search-ok-icon {
  color: #2e8b57;
}

.c0 div svg.search-cancel-icon {
  color: #dc143c;
}

.c1 {
  padding: 0.5em 1.6rem;
  width: 100%;
  border: none;
  border-radius: 20px;
  background: transparent;
}

.c1:focus {
  outline: none;
}

.c3 {
  width: 3rem;
  height: 3rem;
  border: none;
  padding: 0 0.2rem;
  border-left: 2px solid #AAAAAA;
  background: transparent;
  cursor: pointer;
  -webkit-transition: all 350ms ease-in-out;
  transition: all 350ms ease-in-out;
  color: inherit;
}

.c3:hover {
  color: #dc143c;
}

<div>
  <div
    class="c0"
  >
    <div>
      <input
        class="c1"
        name="q"
        placeholder="encontre posts"
        type="search"
        value=""
      />
      <svg
        aria-hidden="true"
        aria-label="Input enabled"
        class="c2 search-ok-icon"
        fill="currentColor"
        focusable="false"
        viewBox="0 0 24 24"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm0 18c-4.41 0-8-3.59-8-8s3.59-8 8-8 8 3.59 8 8-3.59 8-8 8zm4.59-12.42L10 14.17l-2.59-2.58L6 13l4 4 8-8z"
        />
      </svg>
      <button
        class="c3"
      >
        <svg
          aria-hidden="true"
          aria-label="search button"
          class="c2"
          fill="currentColor"
          focusable="false"
          viewBox="0 0 24 24"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M10 18a7.952 7.952 0 0 0 4.897-1.688l4.396 4.396 1.414-1.414-4.396-4.396A7.952 7.952 0 0 0 18 10c0-4.411-3.589-8-8-8s-8 3.589-8 8 3.589 8 8 8zm0-14c3.309 0 6 2.691 6 6s-2.691 6-6 6-6-2.691-6-6 2.691-6 6-6z"
          />
          <path
            d="M11.412 8.586c.379.38.588.882.588 1.414h2a3.977 3.977 0 0 0-1.174-2.828c-1.514-1.512-4.139-1.512-5.652 0l1.412 1.416c.76-.758 2.07-.756 2.826-.002z"
          />
        </svg>
      </button>
    </div>
  </div>
</div>
`;
